app/controllers/middleware_server_controller.rb

```diff
@@ -28,9 +28,47 @@ def listicon_image(item, _view)
     "100/#{icon}.png"
   end
 
+  def button
+    super unless params[:pressed] == 'middleware_server_reload'
+
+    items = params[:miq_grid_checks]
+
+    single_server = identify_record(params[:id])
+    return if single_server.nil? && items.nil? # TODO: re-route somewhere?
+
+    if !items.nil?
+      reload_servers(items)
+    else
+      reload_servers(params[:id]) unless single_server.nil?
+    end
+    render :update do |page|
+      page.replace("flash_msg_div", :partial => "layouts/flash_msg")
+    end
+  end
+
   private ############################
 
   def display_name
     _('Middleware Servers')
   end
+
+  def reload_servers(items)
+    reload_done = false
+    items.split(/,/).each do |item|
+      mw_server = identify_record item
+      if mw_server.product == 'Hawkular'
+        add_flash(_("Not reloading Hawkular server"), :level => :warning)
+      else
+        mw_reload mw_server # TODO: send all requests in one go instead of looping here ?
+        reload_done = true
```

Comment:
Yes. We must not reload the Hawkular server. So I only want to show the flash "triggered reload" if at least one non-Hawkular server was selected.